File: playground/hoa0217/java-springboot-order-perf-optimizer/src/main/java/com/order/perf/domain/Order.java
package com.order.perf.domain;

import com.order.perf.common.BaseEntity;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;
import lombok.Getter;

@Table(name = "ORDERS")
@Entity
@Getter
public class Order extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private int orderNumber; // 주문 번호

    @Column(name = "order_status", nullable = false)
    @Enumerated(EnumType.STRING)
    private OrderStatus orderStatus; // 주문 상태

    protected Order() {
    }

    public Order(final int orderNumber, final OrderStatus orderStatus) {
        this.orderNumber = orderNumber;
        this.orderStatus = orderStatus;
    }
}

File: playground/ruthetum/app/e-commerce/order/src/main/kotlin/team/sipe/spurt/ecommerce/order/domain/Order.kt
package team.sipe.spurt.ecommerce.order.domain

import jakarta.persistence.*
import org.hibernate.annotations.Comment

@Table(name = "orders")
@Entity
class Order(
    @Comment("주문 번호")
    @Column(name = "order_number")
    val orderNumber: Long,

    @Column(name = "order_status")
    @Enumerated(EnumType.STRING)
    val orderStatus: team.sipe.spurt.ecommerce.order.domain.OrderStatus,

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,
): team.sipe.spurt.ecommerce.order.common.BaseEntity()